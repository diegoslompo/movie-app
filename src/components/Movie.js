import React, { Component } from 'react'
import styled from 'styled-components'
import MovieDetails from './MovieDetails'
import Modal from 'react-modal'
import PropTypes from 'prop-types'

class Movie extends Component {

  state = {
    modalId: '',
  }

  static propTypes = {
    modalId: PropTypes.string,
    movieItem: PropTypes.array,
    openModal: PropTypes.func,
    closeModal: PropTypes.func,
  }

  openModal = (id) => { this.setState(() => ({ modalId: id}))}
  closeModal = () => this.setState(() => ({ modalId: '' }))

  render () {

    const { movieItem } = this.props
    const { modalId} = this.state

    return (
      <MovieList className="ds-movie__list">
        {movieItem.map((item) => (
          <Item className="bs-movie__list-item" key={item.imdbID}>
            <LinkItem onClick={() => this.openModal(item.imdbID)}>
              <Image src={item.Poster !== 'N/A' ? item.Poster : 'https://via.placeholder.com/180x270/000'} alt={item.Title} />
            </LinkItem>
            <Modal isOpen={item.imdbID === modalId} >
              <MovieDetails item={item} />
              <button onClick={() => this.closeModal()}> Close </button>
            </Modal>
          </Item>
        ))}
      </MovieList>
    )
  }
} 

const MovieList = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin: 0 -15px;
`;
const Item = styled.div`
  width: 33.333%;
`;
const Image = styled.img`
  height: 100%;
`;
const LinkItem = styled.a`
  display: block;
  margin: 15px;
  height: 270px;
  position: relative;
  cursor: pointer;
  border-radius: 6px;
  overflow: hidden;
  box-shadow: 6px 7px 13px 3px rgba(0, 0, 0, 0.23);

  &:hover::before {
    opacity: 0;
  }

  &::before {
    content: 'view more +';
    position: absolute;
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    background: #4c00e48a;
    display: flex;
    align-items: center;
    justify-content: center;
    color: #fff;
    font-size: 14px;
    font-weight: 900;
    opacity: 1;
    transition: 0.4s;
  }
`;

export default Movie
