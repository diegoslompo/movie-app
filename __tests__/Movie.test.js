import React from "react"
import ShallowRenderer from 'react-test-renderer/shallow';
import Movie from '../src/components/Movie'

describe('Testing component Movie', () => { 
    it('should render corretly', () => {
        const renderer = new ShallowRenderer()
        const wrapper = renderer.render(
            <Movie movieItem={[]} />
        )
        expect(wrapper).toMatchSnapshot()
    })
})