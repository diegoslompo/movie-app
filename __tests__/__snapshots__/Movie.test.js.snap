// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component Movie should render corretly 1`] = `
<ForwardRef(styled.div)
  className="ds-movie__list"
/>
`;
